webapp/src/Entity/Language.php

```diff
@@ -189,6 +193,38 @@ public function getCompileExecutableHash(): ?string
         return $this->compile_executable?->getImmutableExecutable()->getHash();
     }
 
+    #[Serializer\VirtualProperty]
+    #[Serializer\SerializedName('compiler')]
+    #[Serializer\Exclude(if:'object.getCompilerVersionCommand() == ""')]
+    public function getCompilerData(): ?array
+    {
+        $ret = [];
+        if (!empty($this->getCompilerVersionCommand())) {
+            $ret['version_command'] = $this->getCompilerVersionCommand();
+            if (!empty($this->getCompilerVersion())) {
+                $ret['version'] = $this->getCompilerVersion();
+            }
+        } else {
+            return null;
+        }
+        return $ret;
+    }
+
+    #[Serializer\VirtualProperty]
+    #[Serializer\SerializedName('runner')]
+    #[Serializer\Exclude(if:'object.getRunnerVersionCommand() == ""')]
+    public function getRunnerData(): ?array
+    {
+        $ret = [];
+        if (!empty($this->getRunnerVersionCommand())) {
+            $ret['version_command'] = $this->getRunnerVersionCommand();
+            if (!empty($this->getRunnerVersion())) {
+                $ret['version'] = $this->getRunnerVersion();
+            }
+        }
+        return $ret;
```

Comment:
Its unclear to my where there is no `return null` needed here.

Reply:
The `return null` was a previous attempt, now removed above. The `Exclude` with `if:` is the better way to handle it